lms/static/scripts/frontend_apps/components/AutoGradingConfigurator.tsx

```diff
@@ -0,0 +1,215 @@
+  const updatePartialConfig = useCallback(
+    (partialConfig: Partial<AutoGradingConfig>) =>
+      updateAutoGradingConfig({ ...config, ...partialConfig }),
+    [config, updateAutoGradingConfig],
+  );
+
+  const gradingTypeId = useId();
+  const activityCalculationId = useId();
+
+  return (
+    <div className="flex flex-col gap-y-3">
+      <Checkbox
+        checked={active}
+        checkedIcon={CheckboxCheckedFilledIcon}
+        onChange={e =>
+          updatePartialConfig({
+            active: (e.target as HTMLInputElement).checked,
+          })
+        }
+      >
+        Enable automatic participation grading
+      </Checkbox>
+      {active && (
+        <>
+          <div>
+            <h3 id={gradingTypeId} className="font-semibold mb-1">
+              Grading type
+            </h3>
+            <RadioGroup
+              data-testid="grading-type-radio-group"
+              aria-labelledby={gradingTypeId}
+              selected={gradingType}
+              onChange={gradingType => updatePartialConfig({ gradingType })}
+            >
+              <RadioGroup.Radio
+                value="all_or_nothing"
+                subtitle={<small>Must meet minimum requirements.</small>}
+              >
+                All or nothing
+              </RadioGroup.Radio>
+              <RadioGroup.Radio
+                value="scaled"
+                subtitle={<small>Proportional to percent completed.</small>}
+              >
+                Scaled
+              </RadioGroup.Radio>
+            </RadioGroup>
+          </div>
+          <div>
+            <h3 id={activityCalculationId} className="font-semibold mb-1">
+              Activity calculation
+            </h3>
+            <RadioGroup
+              data-testid="activity-calculation-radio-group"
+              aria-labelledby={activityCalculationId}
+              selected={activityCalculation}
+              onChange={activityCalculation =>
+                updatePartialConfig({ activityCalculation })
+              }
+            >
+              <RadioGroup.Radio
+                value="cumulative"
+                subtitle={
+                  <small>Annotations and replies tallied together.</small>
+                }
+              >
+                Calculate cumulative
+              </RadioGroup.Radio>
+              <RadioGroup.Radio
+                value="separately"
+                subtitle={
+                  <small>Annotations and replies tallied separately.</small>
+                }
+              >
+                Calculate separately
+              </RadioGroup.Radio>
+            </RadioGroup>
+          </div>
+          <AnnotationsGoalInputGroup
+            gradingType={gradingType}
+            value={requiredAnnotations}
+            onChange={requiredAnnotations =>
+              updatePartialConfig({ requiredAnnotations })
+            }
+          >
+            Annotations{activityCalculation === 'cumulative' && ' and replies'}
```

Comment:
This kind of string concatenation is OK for the moment, but we'll probably have to change it when we add translation.

Reply:
I could change it to this, which is more readable anyway.

```
{activityCalculation === 'cumulative' ? 'Annotations and replies' : 'Annotations'}